tests/unit/test_serialization.py

```diff
@@ -267,105 +375,167 @@ def assert_almost_equal(value: np.ndarray, reference: Any):
         np.testing.assert_almost_equal(value, reference)
 
 
-def assert_scenario_correct(
-    deserialized_dataset: Mapping[ComponentType, np.ndarray],
+def assert_single_dataset_entries(
```

Comment:
I would be in favor of splitting this function if possible, since several different checks are done at once, it becomes very cognitively complex.